### PR TITLE
(BUGFIX:  plugin-atlassian-auth-provider) pass scope to strategy

### PR DESCRIPTION
**Pull request by postsa**
## Hey, I just made a Pull Request!
Read scopes from auth config and pass them to the passport `AtlassianStrategy`, where [scope is a required option](https://github.com/jsarafajr/passport-atlassian-oauth2/blob/master/lib/strategy.js#L49) to avoid the error message 

```
This app has not requested any supported Atlassian scopes. Check the authorization URL for your app and ensure that it includes valid scopes.
```

image

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/curly-seals-pull.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-auth-backend-module-atlassian-provider': minor
```
Comment (freben) [inline .changeset/curly-seals-pull.md:2]:
```suggestion
'@backstage/plugin-auth-backend-module-atlassian-provider': patch
```

File: plugins/auth-backend-module-atlassian-provider/src/authenticator.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ export const atlassianAuthenticator = createOAuthAuthenticator({
   initialize({ callbackUrl, config }) {
     const clientId = config.getString('clientId');
     const clientSecret = config.getString('clientSecret');
+    const scope = config.getString('scopes');
```
Comment (freben) [inline plugins/auth-backend-module-atlassian-provider/src/authenticator.ts:32]: Could this be `getOptionalString` and have a sensible default value? It must we require every  adopter to set this explicitly?

Also must be added to [the config schema](https://github.com/backstage/backstage/blob/34217c5d377991d05f724a4c01c44e33573e7812/plugins/auth-backend-module-atlassian-provider/config.d.ts#L21)

**Comment by freben** [comment]
Note also that the tests don't pass.

File: plugins/auth-backend-module-atlassian-provider/src/authenticator.ts (hunk 2 above)
Comment (postsa) [reply plugins/auth-backend-module-atlassian-provider/src/authenticator.ts]: Yes good call, done.

**Comment by postsa** [comment]
> Note also that the tests don't pass.

@freben thanks for the quick review! I implemented your suggestions.

File: docs/auth/atlassian/provider.md [hunk 3]
```diff
@@ -46,16 +46,16 @@ auth:
       development:
         clientId: ${AUTH_ATLASSIAN_CLIENT_ID}
         clientSecret: ${AUTH_ATLASSIAN_CLIENT_SECRET}
-        scopes: ${AUTH_ATLASSIAN_SCOPES}
+        scope: ${AUTH_ATLASSIAN_SCOPES}
```
Comment (postsa) [inline docs/auth/atlassian/provider.md:49]: Not opposed to changing this back to `scopes`, but `scope` is consistent with terminology elsewhere.

Comment (freben) [reply docs/auth/atlassian/provider.md:49]: That's fine, but please make the code read both the old and the new name, so you don't break existing users.

Like `config.getOptionalString('scope') ?? config.getOptionalString('scopes') ?? someDefaultValue` or so

Comment (postsa) [reply docs/auth/atlassian/provider.md:49]: Ok, updated.

**Comment by postsa** [comment]
@freben Let me know if there's anything else I should add to this PR before it can be merged. Thanks!

**Review by freben — APPROVED** [review]

**Pull request merged**